### ISSUE TITLE
Conversations Jingle file upload is terribly slow

### ISSUE THREAD

**Issue by marmistrz** [siacs/Conversations#2477]
#### General information

* **Version:** 1.17.1 (a test with 1.18.x will be conducted today in the evening)
* **Device:** OnePlus 3
* **Android Version:** Android 6.0 CM
* **Server name:** wiuwiu.de
* **Conversations source:** F-Droid

#### Steps to reproduce

On wiuwiu.de the max http upload size is 10M. 

1. Share a file with size of 15MB
2. The upload speed is terribly slow

#### Expected result
The upload speed is sensibly fast

#### Debug output

```
05-10 13:32:00.025 18750 18750 D conversations: app switched into foreground
05-10 13:32:04.281 18750 18750 D conversations: app switched into background
05-10 13:34:19.664 18750 18750 D conversations: app switched into foreground
05-10 13:34:32.489 18750 18750 D conversations: app switched into background
05-10 13:34:56.776 18750 18750 D conversations: app switched into foreground
05-10 13:35:00.304 18750 18750 D conversations: app switched into background
05-10 13:36:02.214 18750 18750 D conversations: app switched into foreground
05-10 13:36:06.460 18750 18750 D conversations: app switched into background
05-10 13:36:34.550 18750 18750 D conversations: app switched into foreground
05-10 13:36:39.543 18750 18750 D conversations: app switched into background
05-10 13:38:40.129 18750 22483 D conversations: not auto accepting new file offer with size: 23244712 allowed size:524288
05-10 13:38:40.130 18750 22483 D conversations: update unread count to 1
05-10 13:38:40.255 18750 22483 D conversations: receiving file: expecting size of 23244712
05-10 13:38:46.768 18750 18750 D conversations: app switched into foreground
05-10 13:38:46.792 18750 18750 D conversations: update unread count to 0
05-10 13:39:01.841 18750  8600 D conversations: connection to our own primary candidate failed
05-10 13:39:02.093 18750  8605 D conversations: established connection with proxy.neko.im:7777
05-10 13:39:02.319 18750 22483 D conversations: received candidate error
05-10 13:39:02.320 18750 22483 D conversations: candidate 2qtn2t9gj1 was a proxy. waiting for other party to activate
05-10 13:39:02.484 18750 22483 D conversations: we were responding. receiving file
05-10 13:39:22.214 18750 22483 D conversations: found auth tag as part of omemo key
05-10 13:39:22.216 18750 22483 D conversations: AxolotlService (me@wiuwiu.de):  Received Message with session fingerprint: 05da2ff796a22cf7911c53d6b75d58a99e4c937a3098cfc550e5b1d1cffcc1ee21
05-10 13:39:22.225 18750 22483 D conversations: update unread count to 1
05-10 13:39:22.227 18750 22483 D conversations: me@wiuwiu.de: suppressing notification because conversation is open
05-10 13:39:22.633 18750 18750 D conversations: update unread count to 0
05-10 13:39:22.636 18750 18750 D conversations: me@wiuwiu.de: sending read marker to friend@neko.im/mobile
05-10 13:39:39.534 18750  8607 D conversations: successfully transmitted file:/storage/emulated/0/Conversations/Media/Conversations Files/32380450-7a5b-4bc1-9f82-1fb1b206c4a3_foo.apk (9abd42882fa44bd56cd4ee34191f8c0dee52d2fd)
05-10 13:39:45.705 18750 22483 D conversations: found auth tag as part of omemo key
05-10 13:39:45.707 18750 22483 D conversations: AxolotlService (me@wiuwiu.de):  Received Message with session fingerprint: 05da2ff796a22cf7911c53d6b75d58a99e4c937a3098cfc550e5b1d1cffcc1ee21
05-10 13:39:45.720 18750 22483 D conversations: update unread count to 1
05-10 13:39:45.722 18750 22483 D conversations: me@wiuwiu.de: suppressing notification because conversation is open
05-10 13:39:45.746 18750 18750 D conversations: update unread count to 0
05-10 13:39:45.747 18750 18750 D conversations: me@wiuwiu.de: sending read marker to friend@neko.im/mobile
05-10 13:39:47.112 18750 18750 D conversations: app switched into background
05-10 13:39:47.876 18750 18750 D conversations: app switched into foreground
05-10 13:39:51.563 18750 18750 D conversations: schedule next idle ping
05-10 13:39:51.598 18750 18750 D conversations: AxolotlService (me@wiuwiu.de): Finding devices without session for friend@neko.im
05-10 13:39:51.599 18750 18750 D conversations: AxolotlService (me@wiuwiu.de): Cache miss: b7f2633c-5545-4c79-a661-f4f6418dc71f
05-10 13:39:51.670 18750 19329 D conversations: AxolotlService (me@wiuwiu.de): Generated message, caching: b7f2633c-5545-4c79-a661-f4f6418dc71f
05-10 13:39:51.670 18750 19329 D conversations: AxolotlService (me@wiuwiu.de): Cache hit: b7f2633c-5545-4c79-a661-f4f6418dc71f
05-10 13:39:53.361 18750 18750 D conversations: app switched into background
05-10 13:39:55.321 18750 18750 D conversations: app switched into foreground
05-10 13:40:05.498 18750 18750 D conversations: AxolotlService (me@wiuwiu.de): Finding devices without session for friend@neko.im
05-10 13:40:05.499 18750 18750 D conversations: AxolotlService (me@wiuwiu.de): Cache miss: b7d8a8ec-bc4e-4fb2-a1a4-e2a0c41cdf6f
05-10 13:40:05.569 18750 19329 D conversations: AxolotlService (me@wiuwiu.de): Generated message, caching: b7d8a8ec-bc4e-4fb2-a1a4-e2a0c41cdf6f
05-10 13:40:05.570 18750 19329 D conversations: AxolotlService (me@wiuwiu.de): Cache hit: b7d8a8ec-bc4e-4fb2-a1a4-e2a0c41cdf6f
05-10 13:40:16.649 18750 18750 D conversations: app switched into background
05-10 13:43:37.265 18750 18750 D conversations: app switched into foreground
05-10 13:44:36.048 18750 18750 D conversations: app switched into background
```

### Things that may be related
1. https://www.reddit.com/r/xmpp/comments/3g72gf/running_ejabberd_14_slow_transfers_between/ 
The hypothesis is that Conversations doesn't use the socks bytestream service for transfer.
2. The same thing doesn't happen when a user@neko.im user sends a 20M file to user@wiuwiu.de
3. Both servers support the SOCKS5 bytestream service, but neko.im gives it a subdomain, socks.neko.im, wiuwiu.de not
4. This all was tested behind a NAT.

**Comment by iNPUTmice** [comment]
According to the log the proxy is used. The one from neko.im because the on from wiuwiu doesn't seem to be available (at the time of the log.)
Thus I conclude the proxy is slow.

**Comment by marmistrz** [comment]
Indeed. I tried again and wiuwiu.de -> neko.im is not so slow. But wiuwiu.de -> wiuwiu.de is, and it's not using the proxy:
```
05-10 18:52:27.802  3082  3082 D conversations: me@wiuwiu.de: sending read marker to friend@wiuwiu.de/phone
05-10 18:52:28.092  3082  5388 D conversations: target size overlay: 180.0 overlay bitmap size was 384
05-10 18:52:30.197  3082  3082 D conversations: AxolotlService (me@wiuwiu.de): Finding devices without session for friend@wiuwiu.de
05-10 18:52:30.872  3082  3082 D conversations: app switched into background
05-10 18:52:32.170  3082  3082 D conversations: not all files are under 10485760 bytes. suggesting falling back to jingle
05-10 18:52:32.261  3082  3082 D conversations: app switched into foreground
05-10 18:52:32.261  3082  3082 D conversations: ConversationsActivity.onBackendConnected() - attaching file to conversations. stopping=false
05-10 18:52:32.268  3082 11801 D conversations: processing file as video
05-10 18:52:32.362  3082  3082 D conversations: video transcoding failed This video is not 16:9, and is not able to transcode. (432x240)
05-10 18:52:32.404  3082  3082 D conversations: extracted video dims 432x240
05-10 18:52:32.405  3082  3082 D conversations: me@wiuwiu.de: http upload is not available for files with size 12929050 (max is 10485760)
05-10 18:52:32.405  3082  3082 D conversations: send file message
05-10 18:52:32.405  3082  3082 D conversations: me@wiuwiu.de: http upload is not available for files with size 12929050 (max is 10485760)
05-10 18:52:32.849  3082  3374 D conversations: target size overlay: 180.0 overlay bitmap size was 384
05-10 18:52:34.621  3082  8414 D conversations: me@wiuwiu.de: activating grace period till 18:54:58
05-10 18:52:38.717  3082  8414 D conversations: me@wiuwiu.de: activating grace period till 18:55:02
05-10 18:52:38.746  3082  8414 D conversations: found auth tag as part of omemo key
05-10 18:52:38.748  3082  8414 D conversations: AxolotlService (me@wiuwiu.de):  Received Message with session fingerprint: 05efe74d6ebb09706f7165138ae64b4e8f60afdbabe13020d8d2ed232f5e1de42a
05-10 18:52:38.750  3082  8414 D conversations: me@wiuwiu.de: activating grace period till 18:55:02
05-10 18:52:47.479  3082 12014 D conversations: connection to our own primary candidete failed
05-10 18:52:48.125  3082  8414 D conversations: me@wiuwiu.de: other party received offer
05-10 18:52:50.273  3082  8414 D conversations: found auth tag as part of omemo key
05-10 18:52:50.274  3082  8414 D conversations: AxolotlService (me@wiuwiu.de):  Received Message with session fingerprint: 05df9de824d84a374a0d214a96ab919651464a5335c7da20ac9a700b4e727a9f02
05-10 18:52:50.281  3082  8414 D conversations: update unread count to 1
05-10 18:52:52.413  3082  8414 D conversations: update unread count to 0
05-10 18:52:52.417  3082  8414 D conversations: me@wiuwiu.de: activating grace period till 18:55:16
05-10 18:52:57.326  3082  8414 D conversations: me@wiuwiu.de: activating grace period till 18:55:21
```
